opendevin/controller/state/state.py

```diff
@@ -29,6 +29,7 @@ class State:
     root_task: RootTask = field(default_factory=RootTask)
     iteration: int = 0
     max_iterations: int = 100
+    global_max_iterations: int = 100
```

Comment:
Looking at how this config is being used, I now feel like people could easily misunderstand its usage from its name. They would think this is accumulated max iterations per session, while it's actually max_iterations_per_task. I wonder what people think about it...

Reply:
I think if we change it now, it is better. We can change global_max_iterations to max_iterations_per_task to better reflect its intended use.

Reply:
I completely agree. Per task is much better, it's clear what it's doing.